src/inference/dev_api/threading/ie_executor_manager.hpp

```diff
@@ -21,87 +21,53 @@
-
-private:
-    std::unordered_map<std::string, ITaskExecutor::Ptr> executors;
-    std::vector<std::pair<IStreamsExecutor::Config, IStreamsExecutor::Ptr>> cpuStreamsExecutors;
-    std::mutex streamExecutorMutex;
-    std::mutex taskExecutorMutex;
-};
-
-/**
- * @endcond
- */
-
-/**
- * @brief This is global point for getting task executor objects by string id.
+ * @interface ExecutorManager
+ * @brief Interface for tasks execution manager.
+ * This is global point for getting task executor objects by string id.
  * It's necessary in multiple asynchronous requests for having unique executors to avoid oversubscription.
- * E.g. There 2 task executors for CPU device: one - in GPU, another - in MKLDNN. Parallel execution both of them leads
+ * E.g. There 2 task executors for CPU device: one - in FPGA, another - in MKLDNN. Parallel execution both of them leads
  * to not optimal CPU usage. More efficient to run the corresponding tasks one by one via single executor.
  * @ingroup ie_dev_api_threading
  */
 class INFERENCE_ENGINE_API_CLASS(ExecutorManager) {
 public:
     /**
-     * @brief      Returns a global instance of ExecutorManager
-     * @return     The instance.
+     * A shared pointer to ExecutorManager interface
      */
-    static ExecutorManager* getInstance();
-
-    /**
-     * @brief A deleted copy constructor
-     */
-    ExecutorManager(ExecutorManager const&) = delete;
-
-    /**
-     * @brief A deleted assignment operator.
-     */
-    void operator=(ExecutorManager const&) = delete;
+    using Ptr = std::shared_ptr<ExecutorManager>;
 
     /**
      * @brief Returns executor by unique identificator
      * @param id An unique identificator of device (Usually string representation of TargetDevice)
      * @return A shared pointer to existing or newly ITaskExecutor
      */
-    ITaskExecutor::Ptr getExecutor(std::string id);
+    virtual ITaskExecutor::Ptr getExecutor(const std::string& id) = 0;
 
     /// @private
-    IStreamsExecutor::Ptr getIdleCPUStreamsExecutor(const IStreamsExecutor::Config& config);
+    virtual IStreamsExecutor::Ptr getIdleCPUStreamsExecutor(const IStreamsExecutor::Config& config) = 0;
 
     /**
      * @cond
      */
-    size_t getExecutorsNumber();
+    virtual size_t getExecutorsNumber() const = 0;
 
-    size_t getIdleCPUStreamsExecutorsNumber();
+    virtual size_t getIdleCPUStreamsExecutorsNumber() const = 0;
 
-    void clear(const std::string& id = {});
+    virtual void clear(const std::string& id = {}) = 0;
     /**
      * @endcond
      */
 
-private:
-    ExecutorManager() {}
+    virtual ~ExecutorManager() = default;
 
-    ExecutorManagerImpl _impl;
-
-    static std::mutex _mutex;
-    static ExecutorManager* _instance;
+    /**
+     * @brief      Returns a global instance of ExecutorManager
+     * @return     The instance.
+     */
+    INFERENCE_ENGINE_DEPRECATED("Use IInferencePlugin::executorManager() instead")
+    static ExecutorManager* getInstance();
```

Comment:
can we also create tickets for other plugins who need to migrate to `executorManager()` ?